react/features/base/participants/reducer.js

```diff
@@ -39,6 +39,36 @@ declare var APP: Object;
 const PARTICIPANT_PROPS_TO_OMIT_WHEN_UPDATE
     = [ 'dominantSpeaker', 'id', 'local', 'pinned' ];
 
+/**
+ * Listen for actions which add, remove, or update the set of participants in
+ * the conference.
+ *
+ * @param {Participant[]} state - List of participants to be modified.
+ * @param {Object} action - Action object.
+ * @param {string} action.type - Type of action.
+ * @param {Participant} action.participant - Information about participant to be
+ * added/removed/modified.
+ * @returns {Participant[]}
+ */
+ReducerRegistry.register('features/base/participants', (state = [], action) => {
+    switch (action.type) {
+    case DOMINANT_SPEAKER_CHANGED:
+    case PARTICIPANT_ID_CHANGED:
+    case PARTICIPANT_UPDATED:
+    case PIN_PARTICIPANT:
+        return state.map(p => _participant(p, action));
+
+    case PARTICIPANT_JOINED:
+        return [ ...state, _participant(undefined, action) ];
+
+    case PARTICIPANT_LEFT:
+        return state.filter(p => p.id !== action.participant.id);
```

Comment:
so it can be that PARTICIPANT_LEFT could arrive late, so we need to check if it comes from the right conference

Reply:
(disregard if it's WIP and just split into smaller steps)

Reply:
Correct, the first commit deals with PARTICIPANT_JOINED only. The second commit (which didn't exist and the time you reviewed) deals with PARTICIPANT_LEFT only. The third commit (which doesn't exist at the time I'm writing this comment) will deal with PARTICIPANT_UPDATED only.